workshops/react_calendar/README.md

```diff
@@ -0,0 +1,670 @@
+
+## Part 1: Prerequisites
+
+You should have some familiarity with HTML and JavaScript as well as programming functions like functions, objects, arrays etc. If you know some basics of React, It will be good for you. If not, I will recommend you [this](https://workshops.hackclub.com/nextjs_starter/) workshop. But don't worry, you should be able to follow along regardless. If you're having trouble, feel free to ask [me](https://app.slack.com/client/T0266FRGM/user_profile/U014ND5P1N2) or anyone in the [Hack Club Slack](https://hackclub.com/slack/)!
+
+## Part 2: Setup
+
+So far, we've been using repl.it for most of our workshops. But today, I want to introduce you to another online code editor, [CodeSandbox](codesandbox.io). For making any React projects, CodeSandbox is the best one out there.
+
+To get started, go to this [starter code](https://codesandbox.io/s/calendarstartercode-thk00). Press **`ctrl+s`** / **`cmd+s`** and it will automatically fork it for you. Now, we have everything set up so let's get started!
+
+## Part 3: Building the project
+
+### 1) File Structure and Basics
+
+Let's first have a look at our project's file structure.
+
+First, there are 2 main directories and a package.json file. We'll ignore the package.json file for now and let's have a look at the 2 directories, namely, `public/` and `src/`.
+
+Usually, the `public/` directory contains an HTML file and all your assets. We won't be touching the `public/` directory during the whole workshop, not even the HTML file!
+
+Next is the `src/` directory which contains all your JavaScript files and your CSS files. We have an `index.js` file, which basically renders our React code. Next is the `App.js` file. We can start writing our React code from here itself, but it is a convention to split our code into small components and render them together in our `App.js` file.
+
+This is what our `App.js` file look right now. Let's understand it line by line.
+
+```jsx
+import React from "react";
+import "./styles.css";
+
+export default function App() {
+  return (
+    <div className="App">
+      <h1>Hello CodeSandbox</h1>
+      <h2>Start editing to see some magic happen!</h2>
+    </div>
+  );
+}
+```
+
+At line 1, we import React into our file. Similarly, at line 2, we import the CSS file. We also need to export our `App` component so that the `index.js` file can have an access to it and render it. Therefore, we write our function beginning with `export default`.
+
+Also, this is a React component, so the function will always return jsx code.
+
+**What is jsx?** All the code from line 6 to 9 is known as jsx. It is pretty much similar to HTML but there are few differences which we'll see as we go along.
+
+You'll see that instead of using `class` in our `<div>` tag, we are using `className`, this is one of the difference between jsx and HTML. Whenever we add class to our element in React, we'll use `className` instead of `class`.
+
```

Comment:
The workshop at this point is becoming a bit of a wall of text. Maybe include an image somewhere around here. Anything at all, really—it could be a screenshot of where we're at so far or it could be an unrelated or semi-related gif.